cpp/src/parquet/schema.h

```diff
@@ -141,7 +122,13 @@ class PARQUET_EXPORT Node {
 
   const std::shared_ptr<const LogicalType>& logical_type() const { return logical_type_; }
 
-  int id() const { return id_; }
+  /// \brief The field_id value for the serialized SchemaElement. If the
+  /// field_id is less than 0 (e.g. -1), it will not be set when serialized to
+  /// Thrift
```

Comment:
I'm a bit confused. In other places in this file, "A field_id -1 means to use the default next_id()". Is there a reason for the discrepancy?

Reply:
The `next_id()` default assignments only occur during serialization of the entire `SchemaDescriptor` (I think, will double check), but if you have `field_id == -1` and call `Node::ToParquet` it will happily serialize. I'll add some comments to clarify

Reply:
I fixed these comments. There's follow up investigation to do per PARQUET-1798